persistence/src/test/java/edu/unc/lib/dl/persist/services/transfer/FSToFSTransferClientTest.java

```diff
@@ -163,6 +163,22 @@ public void transferReplaceFileAlreadyExists() throws Exception {
         assertFalse("Source file should not exist after transfer", sourceFile.toFile().exists());
     }
 
+    @Test
+    public void rollbackOnTransferInterruption() throws Exception {
+        String existingContent = "I exist";
+
+        Files.createDirectories(binDestPath.getParent());
```

Comment:
There's actually already a `transferFileAlreadyExists` test which you could extend to check that no extra files are left over. For a test that could fail part way through, from my previous comment
> One test would be to write a file to a directory, then call file.setReadOnly() on that file, then try to overwrite the file (with replace allowed). It should fail partway through, then you will want to verify that the current file still exists, and that there are no extra files in the directory. You'll might need to set file.setWriteable(true) afterwards so that you can delete the file when the test ends, but i haven't verified that.